### ISSUE TITLE
Restore subscriptions with Active Campaign

### ISSUE THREAD

**Issue by reginafcompton** [Metro-Records/la-metro-councilmatic#51]
The RSS feeds do not contain content......

**Comment by reginafcompton** [comment]
Only a problem with the Person RSS feed.

**Comment by shrayshray** [comment]
RSS feeds still aren't working ... they appear on [board report pages](https://boardagendas.metro.net/board-report/2020-0470/), but link to "[Page not found](https://boardagendas.metro.net/board-report/2020-0470/rss/)"
We'd like to have some sort of subscription utility on the site - can we discuss what it would take to enable this, and remove the RSS links until it is launched?

**Comment by hancush** [comment]
Sure thing, @shrayshray – let's chat at our next monthly priorities meeting. Would you like us to escalate the removal of RSS feeds in the meantime? Can take care of that this week, if so. Let me know. :-)

**Comment by shrayshray** [comment]
@hancush yes, please escalate removing the RSS feeds. Thank you!

**Comment by hancush** [comment]
You got it!

**Comment by hancush** [comment]
@shrayshray Removed the links to RSS feeds on board report pages and deployed to production. We can talk more at priorities about a subscription service.

**Comment by hancush** [comment]
Wants / needs of a subscription service:

- Information / context
- Customizable by the user
- Stats on how many are subscribed, and what they're subscribed to (not offered by Legistar)
- Tied to events, e.g., posting of agendas, updates to upcoming agendas, board boxes added (result of particular queryset)

Councilmatic notifications exists, would need an upgrade to be compatible with the board agendas site. I will assess plugin's ability to meet these needs, potentially explore third-party options, as well.

**Comment by hancush** [comment]
Hey, @shrayshray, I did some thinking on approaches for subscriptions:

- **Empower users to create their own subscriptions:** Make RSS feeds more obvious, and direct users to create their own notifications using a third-party service, like Zapier or Feed2Mail
    - Pros: Low cost / level of effort, reliable notifications
    - Cons: No analytics
- **Install the Councilmatic notifications app:** Leverage DataMade's plugin for email subscriptions to actions by people and committees, new bill actions, and search results
    - Pros: Medium cost / level of effort, no third-party integrations (high control and flexibility)
    - Cons: Custom solution, less sophisticated analytics and flexibility of notification mediums
- **Integrate with a third-party marketing service with automated RSS campaigns:** Sign up for a third-party marketing platform, likes MailChimp, and create automated campaigns based off RSS feeds from Councilmatic
    - Pros: Sophisticated analytics, some platforms can send texts, as well as emails
    - Cons: Highest cost and level of effort, highest administrative overhead for Metro

All approaches will require some work, but some will require more than others. Specifically, they all offer the ability to subscribe to _entities_ (people, committees, bills), but not _classes of events_ (any agenda posted, board boxes added). Beyond that, RSS feeds would need to be repaired, per this issue, to support directing users to an external service or integrated with a third-party marketing service. Integration with a third-party marketing service would require still more work to collect and verify user emails for your subscriber lists, and it would need to be administered by you all.

Another differentiating factor is analytics. Directing users to an external service will not give you inside into subscriber behavior, while the notifications app offers very basic reporting, and third-party marketing services offer potentially very detailed analytics, depending on what service you go with.

With all of this in mind, I recommend that we move ahead with the notifications app. One important correction to my previous comment is that our notifications app _will_ work with the upgrade – I 100% forgot we updated it last summer! Out of the box, it offers the ability to subscribe to:

- Sponsorships by a particular person
- Actions by a particular committee
- Meetings of a particular committee 
- Actions on a particular board report
- Results of a particular search

Once we get it installed, we can make improvements to reporting and add additional subscription options, e.g., agendas posted / updated.

But I'm really happy to hear your thoughts, too! What do you think?

**Comment by shrayshray** [comment]
@hancush thank you so much for this write-up! I don't think a 3rd part marketing tool would be cost-prohibitive, and would give Metro more control/customization, etc., but there's also a lot of additional functionality well outside the scope of what we need. Do you happen to have an example or mock up of a notification from the Councilmatic app? I signed up for a few different notifications from Chicago, but not sure when there will be changes to trigger them. Could you also tell us a little more about what analytics are available in the Councilmatic notifications app, how we would view them, and a screenshot of the interface, if possible?
Thank you!

**Comment by hancush** [comment]
Hi, @shrayshray – I just forwarded you and Omar an example of a notification email from the Councilmatic app. 

As I mentioned, the current iteration of the notifications app offers very basic reporting. Essentially, you can view counts and lists of each type of notification:

Listing of subscription types:

![Screen Shot 2020-09-11 at 10 51 35 AM](https://user-images.githubusercontent.com/12176173/92946997-08f93480-f41d-11ea-8ee9-746f48008291.png)

Detail of event subscriptions:

![Screen Shot 2020-09-11 at 10 52 03 AM (2)](https://user-images.githubusercontent.com/12176173/92947788-1236d100-f41e-11ea-8f53-f47c450884f1.png)

Some simple things we could do to make the reporting interface more useful:

- Update the detail views to show email addresses and (where applicable, e.g., search results or a particular person) what the user is subscribed to
- Add useful summary statistics to the detail views
- Wire up functionality to allow you to export subscriber lists

We use [Transactional Email by MailChimp](https://mandrill.zendesk.com/hc/en-us/articles/360047420753-Getting-Started-Guide) (formerly known as Mandrill) to actually send emails. If you wanted to use our central instance, we could generate reports of emails sent for you. You could also set up either your own instance of MailChimp or another SMTP provider we can integrate with (we also like SendGrid, or you may already have an account with another service) and you then would have first-hand access to logs and analytics (things like open rate, click rate, deliverability, etc.)

**Comment by shrayshray** [comment]
@hancush Thank you! Can you confirm our understanding of the subscription types?

Bill action subscriptions: All new and updated bills (includes revisions and minutes histories actions)
Bill search subscriptions: Search results for a keyword or phrase
Committee action subscriptions: All reports with new or updated histories/actions in the Committee 
Committee event subscriptions: New committee meetings, agendas, and updates to the meeting and agenda
Event subscriptions: All new and updated meetings and agendas
Person subscriptions: All reports with new or updated histories/actions in a Committee the Person is a member of 

Also, do you have an idea of the amount of work required to make the updates to the reporting interface? An estimate of hours is fine.

**Comment by hancush** [comment]
@shrayshray – Close, but as written, notifications will generally only be sent for _new_ material, not updated material.

- Bill action: New actions on a particular bill (i.e., new history item on a matter in Legistar) – this does _not_ include updates to the bill itself
- Bill search: New bills matching a particular keyword or phrase
- Committee action: Bills with new actions by a particular committee
- Committee event: New meetings, but not updated meetings, of a particular committee
- Event subscriptions: Ack, looks like this hasn't been implemented on the back end yet – in theory, it would cover new and updated events
- Person subscriptions: New bill sponsorships by a particular person – this doesn't seem useful to Metro, since board reports are sponsored by committees, not people

Given how data entry works on your end (events especially, but also board reports, are updated as a matter of course), it's clear that it's important for a notification service to capture those updates, as well as new material. With that in mind, I think these updates would be useful:

1. Implement the event subscription for new and updated events 
2. Expand the committee meeting subscription to capture new or updated events
3. Add a new bill subscription that allows the user to subscribe to _any_ updates on a particular bill (not just the actions)
4. Expand the bill search notification to capture new or updated bills matching a keyword or phrase
    - I would caution against a subscription to _any_ new or updated bills, since that would be quite noisy for the user. The bill search subscription is meant to help users get notified of bills on topics they care about.

Also note that we can disable subscriptions that you don't want, e.g., person subscriptions (since people don't sponsor bills).

---

In summary, here's what we'd need to do to stand up and improve upon the existing notification integration:

- Install `django-councilmatic-notifications`
- Improve the reporting interface, with some ideas listed in https://github.com/datamade/la-metro-councilmatic/issues/51#issuecomment-691184331
- Update existing or add new subscription options to capture updates to events and board reports

I think that we could accomplish this if we dedicated about month of maintenance time to it (~20 hours), like we did for the online public comment integration.

What do you think?

**Comment by hancush** [comment]
Metro has forwarded docs for ActiveCampaign, will send summary of system requirements. DataMade will look into whether integration is feasible.

**Comment by shrayshray** [comment]
@hancush let's discuss this in our October priorities meeting.

**Comment by hancush** [comment]
I started [a document](https://docs.google.com/document/d/11aQz9Xtd4LpVNSoUx1z1p9DCACp6m7mr0hrv-1QpnXI/edit?usp=sharing) to break this work into smaller pieces. I also have a line out to Active Campaign support re: the best pattern for the workflow we're trying to achieve, since you cannot trigger automations through the API.

**Comment by hancush** [comment]
Metro is going to meet internally to determine priorities, and this will guide how we proceed.

**Comment by shrayshray** [comment]
I noticed the RSS icon is still appearing on the search page: 
![image](https://github.com/Metro-Records/la-metro-councilmatic/assets/28063988/2d1c86b8-591d-4b1d-8903-2d613b175b85)
Can it please be removed from here until we decide next steps?

**Comment by hancush** [comment]
Done in #1034
